Delete dead Tensor code paths

colesbury opened the pull request:
This deletes most of the dead Tensor code paths, including the TensorMethods cwrap and generic/Tensor.cpp.

This also moves the THNN.cwrap/.cpp generation to generate_code which can use ninja if installed.

colesbury committed:
Delete Python Tensor definitions

colesbury committed:
Clean-up THCPModule_initExtension

 * PyObject_SetAttrString does NOT steal a reference
 * fix leak of imported module

colesbury committed:
Delete createTensor() in DynamicTypes

colesbury committed:
Delete THPModule_isTensor

colesbury committed:
Remove stateless functions on _C module

colesbury committed:
Delete _TensorBase

colesbury committed:
Clean-up Module.cpp initialization

colesbury committed:
Fix THCUNN

colesbury committed:
Delete generic Tensor code

 - Delete cwrap code for Tensor and generation
 - Move THNN generation to generate_code.py (use ninja)

colesbury committed:
Fix Python 2.7 build

 - In Python 2.7, the first argument to the module function is null, not
   the module

colesbury committed:
_standard_gamma and _dirichlet_grad are no longer on _C

colesbury committed:
Bind missing sparse functions

colesbury committed:
Delete ModuleSparse.cpp

colesbury committed:
Delete expand_utils

colesbury committed:
Merge _THNN and _THCUNN extensions into _C

gchanan commented on torch/csrc/autograd/functions/init.cpp:
```diff
@@ -110,10 +110,11 @@ bool THPAutograd_initFunctions(PyObject* _unused)
 
-  THPObjectPtr parent(PyImport_ImportModule("torch._C"));
-  if (!parent) return false;
-  PyModule_AddObject(parent.get(), "_functions", module.release());
```
do you need to incref this?

colesbury replied:
I'll add an explicit `Py_INCREF` to make it more clear, but the `module.release()` releases ownership and PyModule_AddObject steals it.

colesbury committed:
Fix Windows test issue

colesbury committed:
Merge branch 'master' into cleanup

colesbury committed:
Explicitly increment module

colesbury commented:
@pytorchbot retest this please

colesbury commented:
@pytorchbot retest this please

colesbury commented:
@pytorchbot retest this please

colesbury committed:
Initialize the THVector dispatch dynamically

This ensures that the THVector dispatch table is loaded when the shared
object is dynamically loaded. This avoids the need for the user to call
THVector_(vectorDispatchInit) which we were only doing in the legacy
Tensor code.

colesbury committed:
Merge branch 'master' into cleanup

colesbury commented:
ASAN build seems flaky today. ONNX build passed earlier

sakaia commented:
It seems "tools/cwrap/plugins/THPPlugin.py" should be removed, since no other module imported "class THPPlugin(CWrapPlugin):"
